### ISSUE TITLE
Any invalid statement after `RUN` appended with `&&` can pass the lint

### ISSUE THREAD

**Issue by shane51** [hadolint/hadolint#760]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
Should throw error if there is invalid statement such as 
```
RUN cd /app && RUN  cd /src
```
or any invalid statement after `RUN` appended with `&&`
### Actual behavior
Pass the lint even there are invalid statement after `RUN` appended with `&&`

### Steps to reproduce the behavior

1. open  https://hadolint.github.io/hadolint/
2. paste the following code and Lint
```
FROM debian:1.0

RUN ./test.sh \
&& invalid_statement

RUN ./test.sh \
&& RUN invalid

RUN ./test.sh \
&& CMD invalid

RUN ./test.sh \
&& apt-get update
```

3. Three invalid statements  passed the lint.

**Output of `hadolint --version` 
 V2.8.0

**Comment by shane51** [comment]
Just checked shellcheck, seems shellcheck's  limitation

https://www.shellcheck.net/

paste the following

```
#! /bin/bash
invalid_statement 
```
passed  shellcheck

**Comment by lorenzo** [comment]
We could add a warning if we find the words `RUN` `COPY` and `CMD` inside a `RUN` instruction... not sure if that would break a lot of docker files in the wild
